### PR TITLE
Add support for Oracle LOOP statement

### PR DESCRIPTION
**Pull request by mvorisek**
docs: https://docs.oracle.com/en/database/oracle/oracle-database/19/lnpls/basic-LOOP-statement.html

### CODE REVIEW

**Comment by greg0ire** [comment]
> Fix fatal error for Oracle LOOP statement 

What fatal error are you talking about?

**Comment by mvorisek** [comment]
```
Doctrine\SqlFormatter\Tests\SqlFormatterTest::testFormat with data set #51 ('BEGIN\n  FOR i IN 1..5\n  LOO...\nEND;', 'BEGIN\n  FOR i IN 1..5 LOOP\n...\nEND;')
ValueError: str_repeat(): Argument #2 ($times) must be greater than or equal to 0

...\src\SqlFormatter.php:87
...\src\SqlFormatter.php:292
...\tests\SqlFormatterTest.php:47
```

(when the tests are run /wo the fix)

**Comment by greg0ire** [comment]
Apparently, situations where the indent level ends up being negative are supposed to be handled by highlighting the last token in red: https://github.com/doctrine/sql-formatter/blob/51e2d67f6ad3813f32dccbd2899c42d732eb656f/src/SqlFormatter.php#L217-L223

Should we address that first?

**Comment by mvorisek** [comment]
> Apparently, situations where the indent level ends up being negative are supposed to be handled by ...
> 
> Should we address that first?

Any warning added to a query is unwanted - it corrupts the query, so yes, this should be addressed, but for 1.4.x by parsing the tokens into matched indentation blocks. If the tokens are unable to be matched (like something is opened but not closed and vice versa), this should be in no indentation, but without any error/warning added to the formatted query.

This PR fixes known `LOOP` not opening an indentation block but `END` closing it. Even with the fix discussed above, we need this PR to format such SQL properly. So it is safe to land this PR now as needed anyway.

**Comment by greg0ire** [comment]
OK so:
- there is a bug, and you are not fixing it here
- support for LOOP is missing, and that is what you are contributing here (so it's a new feature, and you are targeting 1.5.x), and it happens to make a crash disappear, but the same crash can happen in many other situations.

Please rework your commit message to make it more apparent that this is not a fix.

File: src/SqlFormatter.php [hunk 1]
```diff
@@ -285,6 +285,12 @@ public function format(string $string, string $indentString = '  '): string
             } elseif (strtoupper($token->value()) === 'BEGIN') {
                 $newline             = true;
                 $increaseBlockIndent = true;
+            } elseif (strtoupper($token->value()) === 'LOOP') {
```
Comment (greg0ire) [inline src/SqlFormatter.php:288]: Maybe add the link you provided in the description here?

Comment (mvorisek) [reply src/SqlFormatter.php:288]: done

**Comment by mvorisek** [comment]
For the general issue you mentioned I created https://github.com/doctrine/sql-formatter/issues/118 👍

**Review by greg0ire — APPROVED** [review]

**Review by goetas — APPROVED** [review]

**Pull request merged**